assets/javascripts/discourse/services/chat.js

```diff
@@ -545,6 +548,27 @@ export default Service.extend({
     });
   },
 
+  _subscribeToChannelStatusChange() {
+    this.messageBus.subscribe("/chat/channel-status", (busData) => {
+      let activeChannel = this.getActiveChannel();
+      if (
+        busData.chat_channel_id === activeChannel?.id &&
+        (busData.status === CHANNEL_STATUSES.readOnly ||
+          busData.status === CHANNEL_STATUSES.archived)
```

Comment:
Do we do nothing if it's not the active channel? Like visual greying out of the channel or something in the sidebar.

Reply:
I don't think it's worth it, if you open that channel it completely reloads anyway so you will see the new status :) That would fit in with the "fancy reactive channel status change" v2 feature I will make the TODO for